src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -1148,6 +1148,10 @@ private static void MergeHidingLookupResults(LookupResult resultHiding, LookupRe
                             // SPEC: interface type, the base types of T are the base interfaces
                             // SPEC: of T and the class type object. 
 
+                            if (hiddenContainer.Equals(hidingSym.ContainingType, TypeCompareKind.IgnoreNullableModifiersForReferenceTypes))
+                            {
+                                goto symIsHidden; // discard the new candidate, as it is not really different
```

Comment:
>goto symIsHidden [](start = 32, length = 16)

How do we know it is the "same" member? We are not looking at signature etc. For example, in the ```if``` below that is also deals with "hiding" we don't make this assumption. I am assuming that we are "resolving" that "hiding" elsewhere. It feels like the new hiding should be resolved there as well. Meaning that we simply add negated ```if``` condition to the ```if``` below and add additional code elsewhere to handle methods and indexers.

Reply:
It looks like alternatively we could filter out "duplicate" interfaces inside ```GetBaseInterfaces```` helper.

---
In reply to: [524690053](https://github.com/dotnet/roslyn/pull/49338#discussion_r524690053) [](ancestors = 524690053)

Reply:
I don't think we can filter in `GetBaseInterfaces` because that would affect `IsDerivedType` which also uses that helper, but we could do the filtering at the call-site of `GetBaseInterfaces` in `LookupMembersInInterfaceOnly`.
I had done that in drafting this PR, but changed to this approach to reduce allocations and iterations.
Do you have a preference?

I didn't fully understand your question above. We can discuss tomorrow.

---
In reply to: [524709873](https://github.com/dotnet/roslyn/pull/49338#discussion_r524709873) [](ancestors = 524709873,524690053)

Reply:
> I don't think we can filter in GetBaseInterfaces because that would affect IsDerivedType which also uses that helper

That is a fair point. Then it looks like we could filter interfaces in ```LookupMembersInInterfacesWithoutInheritance```. Why bother even performing a lookup if we already looked in a "similar" interface and planning to discard everything we find?

> I didn't fully understand your question above. 

C# hides by signature. Signature is part of detecting if the candidate is different (as the comment says). Where do we check the signature if we are saying "as it is not really different"? If we don't check the signature, we shouldn't claim that "it is not really different". 

---
In reply to: [542952838](https://github.com/dotnet/roslyn/pull/49338#discussion_r542952838) [](ancestors = 542952838,524709873,524690053)